### DISCUSSION TITLE
Persistent player save

### DISCUSSION THREAD

**Discussion by deBearN** [Vdauphin/HeartsAndMinds#1324] [Question and answer]
I was wondering, i saw that in the last update you added some persistent inventory stuff, does Hearts and minds save the player inventory? if not do you think it is compatible with the mod "Persistent RCO", that saves player inventory and stuff added on the map

**Comment by Vdauphin** [comment]
Hello,

> does Hearts and minds save the player inventory?

H&M don't need it because each time you leave the server you spawn back to base with an available Arsenal

Cheers

**Reply by deBearN** [reply]
> Hello,
> 
> > does Hearts and minds save the player inventory?
> > H&M don't need it because each time you leave the server you spawn back to base with an available Arsenal
> 
> Cheers

i asked that because i wanted to make a logistic type of Campaign, like all my playerbase will only have acess to a Arsenal Once, and will have to use crates pre filled with ammo from the eden, which is why i also asked if you know if it's compatible with "Persistent RCO", which is perfect for this kind of campaign

**Reply by Vdauphin** [reply]
I see. There is not. I considered this some years ago but drop the idea as there is many way to get ammo even without Arsenal like:
- creating ammo box at logisitic point
- looting enemy
- also duplicating your inventory by respawning (well now as your body stay on the ground we can disable the inventory save on respawn)

All of this is just mimicking an available Arsenal after all and making the player experience worst.
I prefer to see the resupply gameplay by allowing infinit resources at base so then when players/vehicles are deployed people need to send ammo box, fuel tank, repair truck and ammo supply to the battlefield

Also this kind of thing is not popular as most team already disable the limited Arsenal by role (medic don't have access to machingun for exemple) I made :( . 
Can't see how people will enjoy an Arsenal only available the first time you connect to the campaign

**Reply by ghost** [reply]
I use profilenamespace loadout and position saving loading via a initPlayerLocal.sqf to save player loadouts and locations, seems to be working well so far

**Reply by Vdauphin** [reply]
> I use profilenamespace loadout and position saving loading via a initPlayerLocal.sqf to save player loadouts and locations, seems to be working well so far

Can you elaborate how you play with this? The gameplay you target? How do you deal with the case that a player can connect as soldier and then connect as a medic?

I wrote a script which basically make player slot persistent as vehicles are. So when a player leave the slot (officer, interpreter, medic...) and if an other player connect to the same slot, he will spawn where the previous player was and with the saved inventory, medical state, chemical state... 
For now, I am evaluating it in a gameplay perspective. So any input on that could be usefull

**Reply by ghost** [reply]
> > I use profilenamespace loadout and position saving loading via a initPlayerLocal.sqf to save player loadouts and locations, seems to be working well so far
> 
> Can you elaborate how you play with this? The gameplay you target? How do you deal with the case that a player can connect as soldier and then connect as a medic?
> 
> I wrote a script which basically make player slot persistent as vehicles are. So when a player leave the slot (officer, interpreter, medic...) and if an other player connect to the same slot, he will spawn where the previous player was and with the saved inventory, medical state, chemical state... For now, I am evaluating it in a gameplay perspective. So any input on that could be usefull

I allow every player as interpreter, medic and engineer roles...I host a public server, and so, having the level of coordination required for 1 person per role is limited. 

due to the fact that every player is 'the same', I save the loadouts and locations of players to the profilenamespace so that they can login / logout without interruption, and allows players to do multi day patrols if they so wish.

I still need to find a way to reset the profileNamespace data upon game win/lose, as they'll just spawn where they were, even if it's a different mission file xD
